### DISCUSSION TITLE
Getting HTML code

### DISCUSSION THREAD

**Discussion by mnbh96** [danielsaidi/RichTextKit#177] [Q&A]
Hello. Is there a way to get the HTML code from the editor?

**Comment by martindufort** [comment]
The editor underlying structure is implemented via NSTextView / UITextView which contains a `NSAttributedString`.

You can get that `attributedString` and convert it into HTML by doing:
```
let attributedString = self.richTextEditor.textStorage.attributedString
// Convert into html
let docAttributes = [NSAttributedString.DocumentAttributeKey.documentType: NSAttributedString.DocumentType.html]
do {
	let html = try attributedString.data(from: NSMakeRange(0, attributedString.length), documentAttributes:docAttributes)
	if let htmlString = String(data:html, encoding: .utf8) {
		print(htmlString)
	}
}
catch {
	print("Error unable to convert")
}
```

However be aware that you result will vary given the content of the editor.

**Comment by danielsaidi** [comment]
Wow, that's so cool @martindufort 🤯 

Should this be part of the export feature?

**Reply by martindufort** [reply]
Yes I think that would add value althought I'm not using it. 
Will see If I can do a PR.
